repos/system_upgrade/el8toel9/actors/checkifcfg/libraries/ifcfg.py

```diff
@@ -0,0 +1,145 @@
+        title = 'Rule file detected'
+        summary = ('RHEL 9 does not support the legacy network-scripts'
+                   ' package that was deprecated in RHEL 8 in favor of'
+                   ' NetworkManager. Rule files present in the system will'
+                   ' not work with NetworkManager.')
+        remediation = 'Replace rule files with a NetworkManager dispatcher script.'
+        reporting.create_report([
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Remediation(hint=remediation),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([reporting.Tags.NETWORK, reporting.Tags.SERVICES]),
+            reporting.Flags([reporting.Flags.INHIBITOR]),
+            reporting.RelatedResource('package', 'NetworkManager'),
+            reporting.ExternalLink(
+                title='NetworkManager-dispatcher - Dispatch user scripts for NetworkManager',
+                url='https://networkmanager.dev/docs/api/latest/NetworkManager-dispatcher.html',
+            ),
+        ] + [
+            reporting.RelatedResource('file', fname)
+            for fname in rule_files
+        ])
+
+    if bad_type_files:
+        title = 'Network configuration for unsupported device types detected'
+        summary = ('RHEL 9 does not support the legacy network-scripts'
+                   ' package that was deprecated in RHEL 8 in favor of'
+                   ' NetworkManager. Files for device types that are not'
+                   ' supported by NetworkManager are present in the system.')
+        remediation = ('Consult the nm-settings-ifcfg-rh(5) manual for'
+                       ' valid types of ifcfg files. Remove configuration'
+                       ' files that can not be supported.')
+        reporting.create_report([
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Remediation(hint=remediation),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([reporting.Tags.NETWORK, reporting.Tags.SERVICES]),
+            reporting.Flags([reporting.Flags.INHIBITOR]),
+            reporting.RelatedResource('package', 'NetworkManager'),
+        ] + [
+            reporting.RelatedResource('file', fname)
+            for fname in bad_type_files
+        ])
+
+    if not_controlled_files:
+        title = 'Network configuration with disabled NetworkManager support detected'
+        summary = ('RHEL 9 does not support the legacy network-scripts'
+                   ' package that was deprecated in RHEL 8 in favor of'
+                   ' NetworkManager. Configuration present in the system'
+                   ' prohibit NetworkManager from loading it.')
+        remediation = ('Ensure the ifcfg files comply with format described in'
+                       ' nm-settings-ifcfg-rh(5) manual and remove the'
+                       ' NM_CONTROLLED key from them.')
```

Comment:
Best would be to recommend to migrate to keyfiles and if this is not an option, ensure that the file complies with the nm format.

Reply:
That would be a good idea if we had a migration tool, which we don't have.
Suggesting that people just somehow migrate the configuration without explaining how is just going to do more harm than good.
Also, RHEL 9 supports ifcfg files just fine, there's no need to do this.

This is likely going to come up in future RHEL version though.

Reply:
In this message the users are asked to manually check the ifcfg files for compliance with nm-settings-ifcfg-rh(5). The time spent in doing this seems to be better spent in understanding keyfiles.